scripts/ci/utils/_init.sh

```diff
@@ -0,0 +1,322 @@
+#
+# Output: LOCALLY_BUILT_IMAGES (array)
+function _setup_locally_build_images() {
+    LOCALLY_BUILT_IMAGES=("CI" "CHECKLICENCE")
+    export LOCALLY_BUILT_IMAGES
+}
+
+#
+# Sets up files that should be checked for rebuilding
+# Output: FILES_FOR_REBUILD_CHECK (array)
+function _setup_files_for_rebuild(){
+    FILES_FOR_REBUILD_CHECK=(
+         "setup.py"
+         "setup.cfg"
+         "Dockerfile"
+         ".dockerignore"
+         "airflow/version.py"
+         "airflow/www/package.json"
+         "airflow/www/package-lock.json"
+    )
+    export FILES_FOR_REBUILD_CHECK
+}
+
+#
+# Sets up ports forwarded when entering the container
+# Output: WEBSERVER_HOST_PORT, POSTGRES_HOST_PORT, MYSQL_HOST_PORT
+#
+function _setup_forwarded_ports() {
+    export WEBSERVER_HOST_PORT=${WEBSERVER_HOST_PORT:="28080"}
+    export POSTGRES_HOST_PORT=${POSTGRES_HOST_PORT:="25433"}
+    export MYSQL_HOST_PORT=${MYSQL_HOST_PORT:="23306"}
+}
+
+#
+# Checks if cache is going to be used
+function _check_if_cache_used() {
+    # You can set AIRFLOW_CONTAINER_USE_CACHE to false if you do not want to use standard Docker
+    # cache during build. This way you can test building everything from the scratch
+    AIRFLOW_CONTAINER_USE_CACHE=${AIRFLOW_CONTAINER_USE_CACHE:="true"}
+
+    # You can set AIRFLOW_CONTAINER_USE_LOCAL_DOCKER_CACHE to true if you do not want to use
+    # pulled images at all but you rely on local cache
+    AIRFLOW_CONTAINER_USE_LOCAL_DOCKER_CACHE=${AIRFLOW_CONTAINER_USE_LOCAL_DOCKER_CACHE:="false"}
+}
+
+#
+# Checks if PROD build is enabled
+function _check_if_prod_build_enabled() {
+    # You can set AIRFLOW_CONTAINER_BUILD_PROD_IMAGE to true if you want to enable PROD build
+    # PROD build takes a lot of time so it is disabled by default
+    AIRFLOW_CONTAINER_BUILD_PROD_IMAGE=${AIRFLOW_CONTAINER_BUILD_PROD_IMAGE:="false"}
+}
+
+#
+# Checks if core utils required in the host system are installed and explain what needs to be done if not
+# exits if it's not
+function _check_if_coreutils_installed() {
+    set +e
+    getopt -T >/dev/null
+    local GETOPT_RETVAL=$?
+
+    if [[ $(uname -s) == 'Darwin' ]] ; then
+        command -v gstat >/dev/null
+        local STAT_PRESENT=$?
+    else
+        command -v stat >/dev/null
+        local STAT_PRESENT=$?
+    fi
+
+    command -v md5sum >/dev/null
+    local MD5SUM_PRESENT=$?
+
+    set -e
+
+    local CMDNAME
+    CMDNAME="$(basename -- "$0")"
+
+    ####################  Parsing options/arguments
+    if [[ ${GETOPT_RETVAL} != 4 || "${STAT_PRESENT}" != "0" || "${MD5SUM_PRESENT}" != "0" ]]; then
+        echo
+        if [[ $(uname -s) == 'Darwin' ]] ; then
+            echo >&2 "You are running ${CMDNAME} in OSX environment"
+            echo >&2 "And you need to install gnu commands"
+            echo >&2
+            echo >&2 "Run 'brew install gnu-getopt coreutils'"
+            echo >&2
+            echo >&2 "Then link the gnu-getopt to become default as suggested by brew."
```

Comment:
+1000 for local mac dev instructions :)

Reply:
:D yeah .... Though I am just about to switch to Pixelbook for local development. Keyboard problems and overall stability issues for Mac has gotten much worse